internal/api/core/routing.go

```diff
@@ -34,48 +34,104 @@ type ValidAPIHandler any
 // Post form key for request content.
 const API_REQUEST_CONTENT_KEY = "content"
 
+type Route interface {
+	GetDescription() string
+	GetMethod() string
+	GetRegex() *regexp.Regexp
+	GetSuffix() string
+	Handle(response http.ResponseWriter, request *http.Request) error
+}
+
 // Inspired by https://benhoyt.com/writings/go-routing/
-type Route struct {
-	method  string
-	regex   *regexp.Regexp
-	handler RouteHandler
+type BaseRoute struct {
+	Method      string
+	Suffix      string
+	Regex       *regexp.Regexp
+	Handler     RouteHandler
+	Description string
+}
+
+type APIRoute struct {
+	BaseRoute
+	Request  reflect.Type
+	Response reflect.Type
 }
 
 const MAX_FORM_MEM_SIZE_BYTES = 20 * 1024 * 1024 // 20 MB
 
+func (route *BaseRoute) GetDescription() string {
+	return route.Description
+}
+
+func (route *BaseRoute) GetMethod() string {
+	return route.Method
+}
+
+func (route *BaseRoute) GetRegex() *regexp.Regexp {
+	return route.Regex
+}
+
+func (route *BaseRoute) GetSuffix() string {
+	return route.Suffix
+}
+
+func (route *BaseRoute) Handle(response http.ResponseWriter, request *http.Request) error {
+	return route.Handler(response, request)
+}
+
+func (route *APIRoute) GetDescription() string {
+	return route.Description
+}
+
+func (route *APIRoute) GetMethod() string {
+	return route.Method
+}
+
+func (route *APIRoute) GetRegex() *regexp.Regexp {
+	return route.Regex
+}
+
+func (route *APIRoute) GetSuffix() string {
+	return route.Suffix
+}
+
+func (route *APIRoute) Handle(response http.ResponseWriter, request *http.Request) error {
+	return route.Handler(response, request)
+}
```

Comment:
Doesn't APIRoute already get most of these methods for free since it embeds BaseRoute?